docs(Accessibility): v9 docs pages for accessibility of components and experiences (#20568)
microsoft/fluentui: packages/react-components/src/Concepts/Accessibility/AccessibleExperiences.stories.mdx

import { Meta } from '@storybook/addon-docs';

<Meta title="Concepts/Developer/Accessibility/Experiences" />

## Creating an accessible web application or web page

> When designing an application or a web page, keep in mind that the easiest way to make it accessible is to maintain a consistent and standard compliant document structure.

Reviewer: `standard` doesn't really make sense here

```suggestion
When designing an application or a web page, keep in mind that the easiest way to make it accessible is to maintain a consistent and compliant document structure.
```

> Following steps help to design an accessible user experience:

Reviewer: ```suggestion
The following steps help to design an accessible user experience:
```

> - decompose UI to parts and **identify components, variants and behaviors to use**

Reviewer: This list starts with all lower case characters, while previously they started with upper case, let's follow consistent sentence casing

```suggestion
- Decompose UI to parts and **identify components, variants and behaviors to use**
```

> - define usage of **[headings and landmarks](https://www.w3.org/TR/wai-aria-practices/examples/landmarks/index.html)**

Reviewer: ```suggestion
- define the usage of **[headings and landmarks](https://www.w3.org/TR/wai-aria-practices/examples/landmarks/index.html)**
```

> - verify usage of **[color and contrast](https://webaim.org/resources/contrastchecker/)** to convey information

Reviewer: ```suggestion
- verify the usage of **[color and contrast](https://webaim.org/resources/contrastchecker/)** to convey information
```

> - ensure **tab order and arrow key navigation** is consistent with DOM structure and used ARIA roles

Reviewer: ```suggestion
- ensure **tab order and arrow key navigation** is consistent with DOM structure and ARIA roles
```

> - specify **labels**, especially for components without textual information (icon only buttons) and for containers (lists, toolbars and so on)

Reviewer: ```suggestion
- specify **labels**, especially for components without textual information (e.g. icon only buttons) and for containers (lists, toolbars and so on)
```

- specify texts for **state change announcements** (error messages, confirmations, dynamic UI changes, ...)

> - identify UI parts that **appear on hover or focus** and specify keyboard and screen reader interaction with them. Keep in mind that screen readers and in most cases also touch devices do not support hover state.

Reviewer 2: ```suggestion 
- identify UI parts that **appear on hover or focus** then specify keyboard and screen reader interaction with them. Keep in mind that screen readers and in most cases also touch devices do not support hover state.
```

- list cases when **focus** needs to be **moved programmatically** (if parts of the UI are appearing/disappearing or other cases)

> - list cases when **focus** needs to be **trapped** in sections of the UI (for dialogs and popups or for hierarchical navigation)

Reviewer: ```suggestion
- list cases when **focus** needs to be **trapped** in sections of the UI (for dialogs, popups or hierarchical navigation)
```

> - if extending existing functionality, think about how it fits into current experience with regards to **discoverability, interaction, keyboard navigation and screen reader navigation**

Reviewer: ```suggestion
- when extending existing functionality, think about how it fits into the current experience with regards to **discoverability, interaction, keyboard navigation and screen reader navigation**
```
